kolibri_explore_plugin/collectionviews.py

```diff
@@ -647,7 +650,15 @@ def _create_manifest(grade, name, language):
             _create_manifest(grade, name, language)
 
 
-_read_content_manifests()
+def ensure_initiated(api_function):
+    """Decorator to initiate only once in the first API call."""
+
+    def wrapper(*args, **kwargs):
+        if _collection_download_manager is None:
+            _read_content_manifests()
+        return api_function(*args, **kwargs)
+
+    return wrapper
```

Comment:
Is anything stopping us from memoizing the output of `_read_content_manifests()` so we can move all these globals out of here? Django doesn't have a _built in_ facility for caching generic Python objects … but there is [django-cache-memoize](https://pypi.org/project/django-cache-memoize/), and a little pet memoize decoration in [kolibri.core.notiifcations.utils](https://github.com/learningequality/kolibri/blob/develop/kolibri/core/notifications/utils.py).

Reply:
Memoizing could be part of a follow-up for persisting the collection manifest. Anyways I would keep that in a separate model. My intention was to decouple the collection manifest from the collection state. I wanted (but ran out of time) to come up with an EndlessKeyCollection class that didn't inherit from ContentManifest anymore. Instead:
```python
class EndlessKeyCollection():
    def __init__(self):
        self.state = None # CollectionState object (Django model)
        self.manifest = None # ContentManifest object
```